### PR TITLE
Proof of concept of using client-side script in astro

### PR DESCRIPTION
**Pull request by pkong-ds**
This PR is for demo purpose, will close after @YayunHuang read it.

### CODE REVIEW

File: src/pages/developerTools/_developer-tools-card.ts
Comment (pkong-ds) [inline src/pages/developerTools/_developer-tools-card.ts:1]: Approach 1 
- keep scripts in `src/pages`
- name `ts/js` files using `document` or `window` with `_` prefxi

File: src/components/DeveloperToolCard/developer-tools-card.ts
Comment (pkong-ds) [inline src/components/DeveloperToolCard/developer-tools-card.ts:1]: Approach 2

- Move all astro files into other folders, e.g. `src/componetns`
    - Maybe even better, move all pages to `src/astro-pages/`
        - e.g. `src/astro-pages/DeveloperToolsPage.astro`
        - then, in `src/pages/developerTools/index.astro`, only import the `src/astro-pages/DeveloperToolsPage.astro` w/o any extra logic
        - that way, we can work around `src/pages` problem

rationale: only `src/pages` scripts are buggy

**Review by pkong-ds — COMMENTED** [review]
@YayunHuang think there are 2 approaches I have. See if you can think of others/pick one here?

Also, please make sure lint still work on these ts files 🙏

**Comment by YayunHuang** [comment]
@pkong-ds seems approach 2 won't work too, I tried to put the pages to `src/components` and import the component in `src/pages/xxxx.astro` before and got same error `document is not define`, so let's use approach 1, thanks a lot! :adore2x: :adore2x:

**Comment by pkong-ds** [comment]
As per offline discussion, will use approach 1.

Lemme close this

**Pull request closed without merging**